src/libs/actions/App.js

```diff
@@ -79,14 +87,38 @@ AppState.addEventListener('change', (nextAppState) => {
     appState = nextAppState;
 });
 
-function triggerUpdateAvailable() {
-    Onyx.set(ONYXKEYS.UPDATE_AVAILABLE, true);
+/**
+ * Fetches data needed for app initialization
+ *
+ * @returns {Promise}
+ */
+function getAppData() {
+    NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.NVP_PRIORITY_MODE, 'default');
+    NameValuePair.get(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER, ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, true);
+    getLocale();
+    User.getUserDetails();
+    User.getBetas();
+    User.getDomainInfo();
+    PersonalDetails.fetchLocalCurrency();
+    GeoLocation.fetchCountryCodeByRequestIP();
+    UnreadIndicatorUpdater.listenForReportChanges();
+    BankAccounts.fetchFreePlanVerifiedBankAccount();
+    BankAccounts.fetchUserWallet();
+
+    // We should update the syncing indicator when personal details and reports are both done fetching.
+    return Promise.all([
+        PersonalDetails.fetchPersonalDetails(),
+        Report.fetchAllReports(true, true),
+    ]);
```

Comment:
**thought (non-blocking):** I was kind of expecting to see all of these inside the `Promise.all`. But now wondering whether it matters. Probably the report and personal details are the main things you'd need to "sync".

Reply:
Those two are the only calls returning promises. Not sure if adding the others will make a difference here.

Reply:
Yeah I guess my question could be rephrased as... should these be returning promises and should we wait for them? But not sure it is essential.

Reply:
Yea, I considered that but defaulted to doing nothing since that's the current behavior.

Reply:
In the doc, we said all them would return promises. But not totally necessary imo.